<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Pierre Alliez</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>680</width>
    <height>680</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PWS Rec 2D</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout">
      <item>
       <widget class="GlViewer" name="viewer" native="true">
        <property name="focusPolicy">
         <enum>Qt::StrongFocus</enum>
        </property>
        <property name="locale">
         <locale language="English" country="UnitedStates"/>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QVBoxLayout">
        <item>
         <widget class="QSlider" name="min_mass_slider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout">
          <item>
           <spacer>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QHBoxLayout">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Discard</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="discard_spinbox">
              <property name="maximum">
               <number>10000</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>680</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionLoadPoints"/>
    <addaction name="actionSave"/>
    <addaction name="actionSnapshot"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuPoint_set">
    <property name="title">
     <string>Data</string>
    </property>
    <widget class="QMenu" name="menuPredefined">
     <property name="title">
      <string>Predefined</string>
     </property>
     <addaction name="actionLine"/>
     <addaction name="actionParallel_lines"/>
     <addaction name="actionCircle"/>
     <addaction name="actionHalf_circle"/>
     <addaction name="actionWidely_variable_sampling"/>
     <addaction name="actionSpiral"/>
     <addaction name="actionBox"/>
     <addaction name="actionBoxes"/>
     <addaction name="actionBox_with_boundaries"/>
     <addaction name="actionBox_with_missing_corners"/>
     <addaction name="actionStar"/>
     <addaction name="actionStair"/>
     <addaction name="actionSkyline"/>
     <addaction name="actionIncreasingly_sharp_angles"/>
    </widget>
    <addaction name="menuPredefined"/>
    <addaction name="actionNoise"/>
    <addaction name="actionAdd_outliers"/>
    <addaction name="separator"/>
    <addaction name="actionDecimate"/>
    <addaction name="actionKeep_one_point_out_of_n"/>
    <addaction name="actionSubdivide"/>
    <addaction name="separator"/>
    <addaction name="actionInvert_mass"/>
    <addaction name="actionClamp_mass"/>
    <addaction name="separator"/>
    <addaction name="actionInsertPoint"/>
    <addaction name="separator"/>
    <addaction name="actionClear"/>
   </widget>
   <widget class="QMenu" name="menuAlgorithms">
    <property name="title">
     <string>Algorithms</string>
    </property>
    <widget class="QMenu" name="menuReconstruction">
     <property name="title">
      <string>Decimate</string>
     </property>
     <addaction name="actionReconstruction_one_step"/>
     <addaction name="actionReconstruction_10_steps"/>
     <addaction name="actionReconstruction_100_steps"/>
     <addaction name="actionReconstruction_1000_steps"/>
     <addaction name="actionReconstruction_until"/>
    </widget>
    <addaction name="actionSet_parameters"/>
    <addaction name="actionSet_MChoice"/>
    <addaction name="separator"/>
    <addaction name="actionReconstruction_init"/>
    <addaction name="menuReconstruction"/>
    <addaction name="actionRelocate_vertices"/>
    <addaction name="separator"/>
    <addaction name="actionPrint_Stats"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionView_points"/>
    <addaction name="separator"/>
    <addaction name="actionView_vertices"/>
    <addaction name="actionView_edges"/>
    <addaction name="separator"/>
    <addaction name="actionView_ghost"/>
    <addaction name="actionView_relevance"/>
    <addaction name="actionView_incolors"/>
    <addaction name="separator"/>
    <addaction name="actionView_edge_cost"/>
    <addaction name="actionView_edge_priority"/>
    <addaction name="separator"/>
    <addaction name="actionView_bins"/>
    <addaction name="actionView_foot_points"/>
    <addaction name="actionView_relocation"/>
    <addaction name="actionView_tolerance"/>
    <addaction name="separator"/>
    <addaction name="actionActivate_simulation"/>
    <addaction name="actionView_simulation"/>
    <addaction name="separator"/>
    <addaction name="actionRecenter"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuPoint_set"/>
   <addaction name="menuAlgorithms"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="locale">
    <locale language="English" country="UnitedStates"/>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionClear"/>
   <addaction name="actionLoadPoints"/>
   <addaction name="actionSave"/>
   <addaction name="actionSnapshot"/>
   <addaction name="separator"/>
   <addaction name="actionInsertPoint"/>
   <addaction name="actionActivate_simulation"/>
   <addaction name="separator"/>
   <addaction name="actionRecenter"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionInsertPoint">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset resource="pwsrec.qrc">
     <normaloff>:/icons/inputPoint.png</normaloff>:/icons/inputPoint.png</iconset>
   </property>
   <property name="text">
    <string>Insert mode</string>
   </property>
   <property name="toolTip">
    <string>Insert Point</string>
   </property>
   <property name="statusTip">
    <string>Insert Point</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="icon">
    <iconset resource="pwsrec.qrc">
     <normaloff>:/icons/fileNew.png</normaloff>:/icons/fileNew.png</iconset>
   </property>
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="statusTip">
    <string>Clear</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionLoadPoints">
   <property name="icon">
    <iconset resource="pwsrec.qrc">
     <normaloff>:/icons/fileOpen.png</normaloff>:/icons/fileOpen.png</iconset>
   </property>
   <property name="text">
    <string>Load Points</string>
   </property>
   <property name="statusTip">
    <string>Load Points</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset resource="pwsrec.qrc">
     <normaloff>:/icons/fileSave.png</normaloff>:/icons/fileSave.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="statusTip">
    <string>Save Points</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionCircle">
   <property name="text">
    <string>Circle...</string>
   </property>
  </action>
  <action name="actionHalf_circle">
   <property name="text">
    <string>Half circle...</string>
   </property>
  </action>
  <action name="actionBox">
   <property name="text">
    <string>Box...</string>
   </property>
   <property name="shortcut">
    <string>B</string>
   </property>
  </action>
  <action name="actionLine">
   <property name="text">
    <string>Line...</string>
   </property>
  </action>
  <action name="actionReconstruction_init">
   <property name="text">
    <string>Init</string>
   </property>
   <property name="shortcut">
    <string>I</string>
   </property>
  </action>
  <action name="actionReconstruction_one_step">
   <property name="text">
    <string>1 step</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionView_foot_points">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Foot points</string>
   </property>
   <property name="statusTip">
    <string>View foot points</string>
   </property>
   <property name="shortcut">
    <string>T</string>
   </property>
  </action>
  <action name="actionView_points">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Points</string>
   </property>
   <property name="statusTip">
    <string>View points</string>
   </property>
   <property name="shortcut">
    <string>P</string>
   </property>
  </action>
  <action name="actionView_edges">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Edges</string>
   </property>
   <property name="statusTip">
    <string>View edges</string>
   </property>
   <property name="shortcut">
    <string>E</string>
   </property>
  </action>
  <action name="actionRecenter">
   <property name="icon">
    <iconset resource="pwsrec.qrc">
     <normaloff>:/icons/fit-page-32.png</normaloff>:/icons/fit-page-32.png</iconset>
   </property>
   <property name="text">
    <string>Recenter</string>
   </property>
  </action>
  <action name="actionView_vertices">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Vertices</string>
   </property>
   <property name="shortcut">
    <string>V</string>
   </property>
  </action>
  <action name="actionBoxes">
   <property name="text">
    <string>Two boxes...</string>
   </property>
  </action>
  <action name="actionStair">
   <property name="text">
    <string>Stair...</string>
   </property>
  </action>
  <action name="actionSkyline">
   <property name="text">
    <string>Skyline...</string>
   </property>
  </action>
  <action name="actionView_edge_priority">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Edge priority</string>
   </property>
   <property name="shortcut">
    <string>Z</string>
   </property>
  </action>
  <action name="actionReconstruction_10_steps">
   <property name="text">
    <string>10 steps</string>
   </property>
   <property name="shortcut">
    <string>1</string>
   </property>
  </action>
  <action name="actionReconstruction_100_steps">
   <property name="text">
    <string>100 steps</string>
   </property>
   <property name="shortcut">
    <string>2</string>
   </property>
  </action>
  <action name="actionReconstruction_1000_steps">
   <property name="text">
    <string>1000 steps</string>
   </property>
   <property name="shortcut">
    <string>3</string>
   </property>
  </action>
  <action name="actionAdd_outliers">
   <property name="text">
    <string>Add outliers</string>
   </property>
   <property name="shortcut">
    <string>O</string>
   </property>
  </action>
  <action name="actionSnapshot">
   <property name="icon">
    <iconset resource="pwsrec.qrc">
     <normaloff>:/icons/snapshot.png</normaloff>:/icons/snapshot.png</iconset>
   </property>
   <property name="text">
    <string>Snapshot</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actionIncreasingly_sharp_angles">
   <property name="text">
    <string>Increasingly sharp angles...</string>
   </property>
  </action>
  <action name="actionBox_with_boundaries">
   <property name="text">
    <string>Box with boundaries...</string>
   </property>
  </action>
  <action name="actionBox_with_missing_corners">
   <property name="text">
    <string>Box with missing corners...</string>
   </property>
  </action>
  <action name="actionStar">
   <property name="text">
    <string>Star...</string>
   </property>
  </action>
  <action name="actionSpiral">
   <property name="text">
    <string>Spiral...</string>
   </property>
  </action>
  <action name="actionSet_parameters">
   <property name="text">
    <string>Parameters</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionView_edge_cost">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Edge cost</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="actionReconstruction_until">
   <property name="text">
    <string>until</string>
   </property>
   <property name="shortcut">
    <string>U</string>
   </property>
  </action>
  <action name="actionParallel_lines">
   <property name="text">
    <string>Parallel lines...</string>
   </property>
  </action>
  <action name="actionNoise">
   <property name="text">
    <string>Noise</string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
  </action>
  <action name="actionActivate_simulation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="pwsrec.qrc">
     <normaloff>:/icons/vertex.png</normaloff>:/icons/vertex.png</iconset>
   </property>
   <property name="text">
    <string>Simulation</string>
   </property>
  </action>
  <action name="actionView_simulation">
   <property name="text">
    <string>Simulation stage</string>
   </property>
   <property name="shortcut">
    <string>A</string>
   </property>
  </action>
  <action name="actionRelocate_vertices">
   <property name="text">
    <string>Relocate</string>
   </property>
   <property name="shortcut">
    <string>L</string>
   </property>
  </action>
  <action name="actionView_relocation">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Relocation</string>
   </property>
   <property name="shortcut">
    <string>Shift+L</string>
   </property>
  </action>
  <action name="actionView_ghost">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Ghost edges</string>
   </property>
   <property name="shortcut">
    <string>G</string>
   </property>
  </action>
  <action name="actionInvert_mass">
   <property name="text">
    <string>Invert mass</string>
   </property>
   <property name="shortcut">
    <string>Shift+I</string>
   </property>
  </action>
  <action name="actionView_relevance">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Relevance</string>
   </property>
   <property name="shortcut">
    <string>Shift+R</string>
   </property>
  </action>
  <action name="actionView_tolerance">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Tolerance</string>
   </property>
   <property name="shortcut">
    <string>Shift+T</string>
   </property>
  </action>
  <action name="actionView_incolors">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>In colors</string>
   </property>
  </action>
  <action name="actionClamp_mass">
   <property name="text">
    <string>Clamp mass</string>
   </property>
  </action>
  <action name="actionView_bins">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Bins</string>
   </property>
   <property name="shortcut">
    <string>Shift+B</string>
   </property>
  </action>
  <action name="actionPrint_Stats">
   <property name="text">
    <string>Print Stats</string>
   </property>
   <property name="shortcut">
    <string>Shift+S</string>
   </property>
  </action>
  <action name="actionSubdivide">
   <property name="text">
    <string>Subdivide</string>
   </property>
  </action>
  <action name="actionWidely_variable_sampling">
   <property name="text">
    <string>Widely variable sampling</string>
   </property>
  </action>
  <action name="actionDecimate">
   <property name="text">
    <string>Decimate</string>
   </property>
  </action>
  <action name="actionKeep_one_point_out_of_n">
   <property name="text">
    <string>One point out of n</string>
   </property>
  </action>
  <action name="actionSet_MChoice">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Multiple Choice</string>
   </property>
   <property name="shortcut">
    <string>M</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GlViewer</class>
   <extends>QWidget</extends>
   <header>glviewer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>min_mass_slider</tabstop>
  <tabstop>discard_spinbox</tabstop>
 </tabstops>
 <resources>
  <include location="pwsrec.qrc"/>
 </resources>
 <connections/>
</ui>
